src/drivers/common/memory.h

```diff
@@ -16,10 +16,14 @@ struct periph_memory_desc {
 	uint32_t len;
 };
 
+#define MAX_SEGMENTS 64
```

Comment:
Since you made this macro global, you should assign a unique name to it. Let's call it PERIPH_MAX_SEGMENTS, for example.